services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -622,6 +643,141 @@ testBrigSideIsUpdated = do
+                [ replaceAttrib "externalId" externalId ]
+            let user'' = Scim.value . Scim.thing $ storedUser'
+            liftIO $ Scim.User.externalId user'' `shouldBe` externalId
+        it "replacing every supported atttribute at once works" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            user' <- randomScimUser
+            storedUser <- createUser tok user
+            let userid = scimUserId storedUser
+            let externalId = Scim.User.externalId user'
+            let userName = Scim.User.userName user'
+            let displayName = Scim.User.displayName user'
+            storedUser' <- patchUser tok userid $ PatchOp.PatchOp
+                [ replaceAttrib "externalId" externalId
+                , replaceAttrib "userName" userName
+                , replaceAttrib "displayName" displayName
+                ]
+            let user'' = Scim.value . Scim.thing $ storedUser'
+            liftIO $ Scim.User.externalId user'' `shouldBe` externalId
+            liftIO $ Scim.User.userName user'' `shouldBe` userName
+            liftIO $ Scim.User.displayName user'' `shouldBe` displayName
+
+        -- NOTE: Also the "extra" attributes are ignored 
+        --
+        it "other valid attributes are quietly ignored for now" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let userid = scimUserId storedUser
+            storedUser' <- patchUser tok userid $ PatchOp.PatchOp
+                [ replaceAttrib "emails" ("hello" :: Text) ]
+            liftIO $ storedUser `shouldBe` storedUser'
+        it "invalid attributes are quietly ignored for now" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let userid = scimUserId storedUser
+            storedUser' <- patchUser tok userid $ PatchOp.PatchOp
+                [ replaceAttrib "totallyBogus" ("hello" :: Text) ]
+            liftIO $ storedUser `shouldBe` storedUser'
+
+        -- NOTE: Remove at the moment actually never works! As all the fields
+        -- we support are required in our book
+        it "userName cannot be removed according to scim" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let userid = scimUserId storedUser
+            _ <- patchUser tok userid $ PatchOp.PatchOp
+                [ removeAttrib "userName" ]
+            pure ()
+        it "displayName cannot be removed in spar (though possible in scim). Diplayname is required in Wire" $ do
+            (tok, _) <- registerIdPAndScimToken
+            user <- randomScimUser
+            storedUser <- createUser tok user
+            let userid = scimUserId storedUser
+            _ <- patchUser tok userid $ PatchOp.PatchOp
```

Comment:
This seems to currently return a 200, right? According to [the spec](https://tools.ietf.org/html/rfc7644#section-3.12) should this not return a 400?